spec/observables/from-spec.ts

```diff
@@ -232,11 +34,71 @@ describe('Observable.from', () => {
     });
   });
 
-  it('should return T and map for arrays', () => {
+  it('should return T for arrays', () => {
     type(() => {
       /* tslint:disable:no-unused-variable */
-      let o1: Rx.Observable<number> = Observable.from(<number[]>[], x => x.toString(), null, Rx.Scheduler.asap);
+      let o1: Rx.Observable<number> = Observable.from(<number[]>[], Rx.Scheduler.asap);
       /* tslint:enable:no-unused-variable */
     });
   });
+
+  const fakerator = (...values) => ({
+    [<symbol>Symbol.iterator]: () => {
+      const clone = [...values];
+      return {
+        next: () => ({
+          done: clone.length <= 0,
+          value: clone.shift()
+        })
+      };
+    }
+  });
+
+  const sources: { name: string, value: any }[] = [
+    { name: 'observable', value: Observable.of('x') },
+    { name: 'array', value: ['x'] },
+    { name: 'promise', value: Promise.resolve('x') },
+    { name: 'iterator', value: fakerator('x') },
+    { name: 'array-like', value: { [0]: 'x', length: 1 }},
+    { name: 'string', value: 'x'},
+    { name: 'arguments', value: function(x) { return arguments; }('x') },
```

Comment:
This is only missing one for `[Symbol.observable]()`... which should return an object with a `subscribe(observer: Observer<T>): Unsubscribable`